components/automate-cli/pkg/diagnostics/integration/iam_v2_policies.go

```diff
@@ -139,29 +164,63 @@ func CreateIAMV2PoliciesDiagnostic() diagnostics.Diagnostic {
 				return err
 			}
 
-			tstCtx.SetValue("iam-policies-v2", iamV2PoliciesSave{
+			loaded := generatedV2PolicyData{}
+			err = tstCtx.GetValue("iam-policies-v2", &loaded)
+			if err != nil {
+				return errors.Errorf(err.Error(), "could not find generated context")
+			}
+
+			tstCtx.SetValue("iam-policies-v2", generatedV2PolicyData{
 				TokenID:    pol.TokenID,
 				TokenValue: tokenInfo.Value,
 				PolicyID:   policyInfo.ID,
 				TeamID:     pol.TeamID,
 				UserID:     pol.UserID,
 				RoleID:     pol.RoleID,
 				ProjectID:  pol.ProjectID,
+				Skipped:    loaded.Skipped,
 			})
 			return nil
 		},
 		Verify: func(tstCtx diagnostics.VerificationTestContext) {
-			loaded := iamV2PoliciesSave{}
-			err := tstCtx.GetValue("iam-policies-v2", &loaded)
-			require.NoError(tstCtx, err, "Could not load generated context")
+			// in case there was an upgrade, check for v1 policy data
+			v1loaded := iamV1PoliciesSave{}
+			err := tstCtx.GetValue("iam-policies-v1", &v1loaded)
```

Comment:
If Generate was run while on v1, we might have some v1 policies migrated to v2 that we can verify.